packages/api-server/api_server/repositories/alerts.py

```diff
@@ -71,13 +72,23 @@ async def acknowledge_alert(self, alert_id: str) -> Optional[ttm.AlertPydantic]:
         # https://github.com/tortoise/tortoise-orm/pull/1131. This is a
         # temporary workaround.
         ack_alert._custom_generated_pk = True  # pylint: disable=W0212
+        unix_millis_acknowledged_time = round(ack_time.timestamp() * 1e3)
         ack_alert.update_from_dict(
             {
                 "acknowledged_by": self.user.username,
-                "unix_millis_acknowledged_time": round(ack_time.timestamp() * 1e3),
+                "unix_millis_acknowledged_time": unix_millis_acknowledged_time,
             }
         )
         await ack_alert.save()
+
+        user = User(username="__rmf_internal__", is_admin=True)
+        task_repo = TaskRepository(user)
+
+        # Save in logs who was the user that acknowledged the task
+        await task_repo.save_log_acknowledged_task_completion(
```

Comment:
```suggestion
        await self.task_repo.save_log_acknowledged_task_completion(
```
with the suggestion in the main review comment, we can remove the construction of `task_repo` in the few lines before this, and just use `self.task_repo`

Reply:
Oh, man! Thanks so much for catching it. 
Unnecessary construction. 
Done 8b334f4cc27f7722a55acd08af8a57bee260ce03

Reply:
`Self` here is `AlertRepository` and I don't have access to `task_repo`. What I could do is `self.user` but I make the instance of `User` to make sure it is admin.

Reply:
oh! `self.task_repo` would refer to the `TaskRepository` that is passed in during construction, take a look at the example code block I provided in the main review comment

```
class AlertRepository:
    def __init__(self, user: User, task_repo: TaskRepository = None):
        self.user = user
        self.task_repo = task_repo if task_repo is not None else TaskRepository(self.user)
```

Reply:
`task_repo` can be passed into `alert_repo` here, https://github.com/open-rmf/rmf-web/blob/feature/acknowledged_by-showed-in-task-logs/packages/api-server/api_server/routes/internal.py#L15

Reply:
Gotcha!
Sorry for my misunderstanding! 
Done here! c2cd6e7fb0cabd00c586ce3a9cdf820a56508c6a